packages/xchain-radix/src/client.ts

```diff
@@ -0,0 +1,465 @@
+   * @param {string} txId The transaction id.
+   * @returns {Tx} The transaction details of the given transaction id.
+   */
+  async getTransactionData(txId: string): Promise<Tx> {
+    const url = `${this.getGatewayUrl()}${TRANSACTION_COMMITTED_DETAILS_PATH}`
+    const requestBody = {
+      intent_hash: txId,
+      opt_ins: {
+        manifest_instructions: true,
+        raw_hex: true,
+      },
+    }
+
+    try {
+      const response = await axios.post(url, requestBody)
+      const transaction: Tx = await this.convertTransactionFromHex(
+        response.data.transaction.raw_hex,
+        response.data.transaction.confirmed_at,
+        response.data.transaction.intent_hash,
+      )
+      return transaction
+    } catch (error) {
+      throw new Error('Failed to fetch transaction data')
+    }
+  }
+
+  /**
+   * Helper function to convert a transaction in hex, returned by the gateway to a Tx type
+   * @param transaction_hex - The raw_hex returned by the gateway for a transaction id
+   * @param confirmed_at - The confirmed_at date for the transaction
+   * @param intent_hash - The transaction intent hash
+   * @returns a transaction in Tx type
+   */
+  async convertTransactionFromHex(transaction_hex: string, confirmed_at: string, intent_hash: string): Promise<Tx> {
+    const binaryString = Buffer.from(transaction_hex, 'hex').toString('binary')
+    const transactionBinary = new Uint8Array(binaryString.split('').map((char) => char.charCodeAt(0)))
+    const transactionSummary = await LTSRadixEngineToolkit.Transaction.summarizeTransaction(transactionBinary)
+    const withdrawAccount = Object.keys(transactionSummary.withdraws)[0]
+    const depositAccount = Object.keys(transactionSummary.deposits)[0]
+    const withdrawResource = Object.keys(transactionSummary.withdraws[withdrawAccount])[0]
+    const withdrawAmount: number = transactionSummary.withdraws[withdrawAccount][withdrawResource].toNumber()
+
+    const transaction: Tx = {
+      from: [
+        {
+          from: withdrawAccount,
+          amount: baseAmount(withdrawAmount),
+          asset: { symbol: withdrawResource, ticker: withdrawResource, synth: false, chain: 'radix' },
+        },
+      ],
+      to: [
+        {
+          to: depositAccount,
+          amount: baseAmount(withdrawAmount),
+          asset: { symbol: withdrawResource, ticker: withdrawResource, synth: false, chain: 'radix' },
+        },
+      ],
+      date: new Date(confirmed_at),
+      type: TxType.Transfer,
+      hash: intent_hash,
+      asset: { symbol: withdrawResource, ticker: withdrawResource, synth: false, chain: 'radix' },
+    }
+    return transaction
+  }
+
+  /**
+   * Creates a transaction using the SimpleTransactionBuilder, signs it with the
+   * private key and returns the signed hex
+   * @param params - The transactions params
+   * @returns A signed transaction hex
+   */
+  async transfer(params: TxParams): Promise<string> {
```

Comment:
Note to self - raise discussion point about how they'll transfer to MayaRouter using this. Perhaps if the `toAccount` is a known component address (e.g. the MayaRouter), we might need to build a different transaction.